### PR TITLE
KAFKA-17609:[3/4]Convert system tests to kraft part 3

### PR DESCRIPTION
**Pull request by bbejeck**
1. Broker compatibility test
3. [ Streams optimized test]( https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=c2fe7c00-578e-38da-812a-0f3d0661ab68)
4. [Streams relational smoke test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=1f108b88-a76e-3cd1-add8-c5958cac7bd0)
5. [Streams shutdown deadlock test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=3b955bc9-bb36-3517-b8c7-ccfeae681926)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
Links to passing tests
[Streams optimized test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=c2fe7c00-578e-38da-812a-0f3d0661ab68)
[Streams relational smoke test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=1f108b88-a76e-3cd1-add8-c5958cac7bd0)
[Streams shutdown deadlock test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=3b955bc9-bb36-3517-b8c7-ccfeae681926)

Broker compatibility test is failing but seems unrelated to the changes for switching from ZK to KRaft

File: tests/kafkatest/tests/streams/streams_relational_smoke_test.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def __init__(self, test_context):
 
     @cluster(num_nodes=8)
     @matrix(crash=[False, True],
-            processing_guarantee=['exactly_once', 'exactly_once_v2'],
+            processing_guarantee=['exactly_once_v2'],
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_relational_smoke_test.py]: Given that we have only EOSv2 left, should we not just remove this parameter completely?

Given that `exactly_once` was remove from the code base a while ago already, I am wondering why we did not get any failing system test alerts? (Miss on my side to not remove this setting in one of my PRs) -- Or is this test not executed? 🤔

File: tests/kafkatest/tests/streams/streams_relational_smoke_test.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def __init__(self, test_context):
...
             metadata_quorum=[quorum.isolated_kraft])
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_relational_smoke_test.py]: Seem we can remove this parameter, too, and also switch to `combined_kraft` ?

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 3]
```diff
@@ -18,34 +18,27 @@
...
 
 
 class StreamsBrokerCompatibility(Test):
     """
     These tests validates that
-    - Streams works for older brokers 0.11 (or newer)
-    - Streams w/ EOS-alpha works for older brokers 0.11 (or newer)
     - Streams w/ EOS-v2 works for older brokers 2.5 (or newer)
-    - Streams fails fast for older brokers 0.10.0, 0.10.2, and 0.10.1
-    - Streams w/ EOS-v2 fails fast for older brokers 2.4 or older
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: I am not sure why we would need any changes here? For compatibility testing, we should still be able to startup old Kafka with ZK and nothing should really change?

Of course, when using `trunk/4.0` we would start brokers in kraft mode.

Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: My understanding was that all ZK was going to be completely removed maybe we can go with this as is and in a follow-up PR revisit and see what needs to be added back

Comment (mjsax) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: Let's clarify with @cmccabe before we merge this.

File: tests/kafkatest/tests/streams/streams_relational_smoke_test.py (hunk 2 above)
Comment (mjsax) [reply tests/kafkatest/tests/streams/streams_relational_smoke_test.py]: Seems you missed this one?

**Review by mjsax — APPROVED** [review]
LGTM

**Comment by bbejeck** [comment]
[Link to passing ](https://semaphore.ci.confluent.io/jobs/9fa811f8-b6aa-4eba-a366-8def6c28f7ca/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976)tests.

zk_to_kraft_conversion_pr_3

**Comment by bbejeck** [comment]
Merged #17327 into trunk

**Pull request merged**